### PR TITLE
Validate YAML file using schema before constructing module.

### LINKED ISSUE #1523: Validate YAML using schema
### Addressing a Problem?

Help identify errors in YAML indicator definition by validating config against schema. 

### Potential Solution

_No response_

### Additional context

_No response_

### Contribution

- [ ] I would be willing/able to open a Pull Request to contribute this feature.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1523
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] CHANGES.rst has been updated (with summary of main changes)
  - [ ] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Use Yamale to validate schema before constructing module. Good news, no error found in the test suite. 

### Other information:

There was an issue raised by @tlogan2000 about indicators accepting wrong arguments without failing. Could you post an offending example ?

### CODE REVIEW

File: CHANGES.rst
Comment (Zeitsperre) [inline CHANGES.rst:5]:
```suggestion
v0.48.0 (unreleased)
--------------------
Contributors to this version: David Huard (:user:`huard`),

New features and enhancements
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
* Validate YAML indicators description before trying to build module. (:issue:`1523`)
```

Comment (huard) [reply CHANGES.rst:5]: Yeah, I knew I'd have to fiddle there after the merge of the uncertainty partitioning method, so didn't bother.

**Review by aulemahal — APPROVED** [review]
Looks great!

I first thought that the other issue raised by @tlogan2000  was for yamls like this:
```yaml
realm: atmos
indicators:
  tg:
    base: tg_mean
    parameters:
      frq: MS
```
where "frq" is misspelt. However, I just tested and this raises an error. So either the original error was fixed elsewhere, or the error was with the keys a level above, and this PR fixes it.

**Comment by huard** [comment]
Did you test it with or without this PR ?

**Comment by aulemahal** [comment]
With and without.

```yaml
realm: atmos
indicators:
  tg:
    base: tg_mean
    parameters:
      frq: MS
```
fails on both, while
```yaml
realm: atmos
indicators:
  tg:
    base: tg_mean
    parametrs:
      freq: MS
```
fails only here.

**Comment by huard** [comment]
Ok, let's merge this then and see if anything else comes up.

**Comment by huard** [comment]
Should we worry about the error in test_encoding ? This occurs in icclim.yml

```
UnicodeDecodeError: 'gbk' codec can't decode byte 0xa5 in position 4213: illegal multibyte sequence
```

**Comment by huard** [comment]
C'est "≥" qui cause problème.

**Pull request merged**
